### DISCUSSION TITLE
Concept & Design : sponsor's settlement influence

### DISCUSSION THREAD

**Discussion by mokun** [mars-sim/mars-sim#425] [Ideas]
The exploration of Mars is not only a grand engineering challenge, it is also expensive. Only nation stats could only afford to pay for such a grand undertaking.

However, this picture has changed somewhat over the recent years. *MarsOne* was one of the first attempt to ignite the imagination of common citizens about making Mars the humanity's second home. 

With the rise of space start-up in rocket launches and small sats and cub-sats, Mars is getting ready to open up for business soon. 

In mars-sim, there are 9 sponsoring space agencies/organization. See [wiki](https://github.com/mars-sim/mars-sim/wiki/Sponsorship)

The rule of thumb is that each settlement needs a sponsor. At the same time, each person has a country of origin and also a sponsor.

e.g. If NASA is the sponsor of the Plymouth Base, then at the start of the sim, all of its settlers would have USA as the country of origin.

Since ESA is a multi-national space agency, the settlement it sponsors will have settlers coming from 22 nations.

The Mars Society is an international organization. If being the sponsor of a settlement, its settlers will come from all 28 possible nations.

SpaceX is a USA based commercial venture. So the settlers of the settlement it sponsors will be from USA.

What we have not modeled and put to use is how these sponsors should impact the settlement and the day-to-day decisions of the settlers on this thread.

Q1 : what can we envision about the role of a sponsor ?

Q2 : how should should impact the mission planning over a settlement ? 

Q3 : how should the leadership interface with its sponsor on Earth and how should the sponsor affect the leadership ? 

Q4 : what kind of resources or supplies should a sponsor provide for an initial settlement ? 

Feel free to comment and offer feedback so that we may put together design idea and formulate programming models for the use of sponsors in mars-sim.

**Comment by Urwumpe** [comment]
My two cents there:

Q1: I see the following possible roles of a sponsor in the simulation:

- Provides free resupply from Earth (Basic services)
- Allows to buy special goods from them in exchange for more influence of the sponsor on Mars
- Could insert special satellites and probes into Mars Orbit or land robotic probes on Mars (for example: Better communication links to Earth, or communication even in conjunction)
- Could operate large scale infrastructure in Space like Space Stations or Aldrin Cyclers
- Trains and selects new settlers from Earth (Nationality, Skills, Knowlegde)
- Gives objectives to individual settlers or organisations
- Maintains relations between other sponsors - e.g. SpaceX could cooperate with NASA or ESA, but also break the ties should this no longer be in agreement with their strategy. 
- Maybe we could allow sponsors to buy settlements or installations on Mars from other sponsors. Or trade shares of a commercial base. 

We could maybe also allow "invisible" sponsors like criminal organisations or intelligence agencies. 

Q2: By the objectives that a sponsor might give to an organisation, may it be a full settlement or a team of scientists or engineers. An objective could be simple ("Produce 100 gaskets") or a full project that the group should achieve ("Install a well at site xyz"). Also, a sponsor could create demand for special luxury goods among the settlers ("I want a steak!") and thus cause pressure on the command from below. 

Q3: Initially just by the commander, maybe later sponsor diplomacy can be delegated to a special diplomat role. There should be a way to send requests to sponsors and a way to get objectives and requests from sponsors. 

Q4: Mostly those that can't be easily produced on Mars, also for maintaining some pressure on the Mars settlers to interact with them. For example, frozen meat would be some rare resource on Mars. But also complex machines like reactors or Sabatier processors. All sponsors don't want to be just spectators, they will all try to get influence on Mars by providing things that Mars lacks.

**Comment by mokun** [comment]
> Allows to buy special goods from them in exchange for more influence of the sponsor on Mars

Yes. We could set up a running sponsor influence point system. 

It's sort of keeping track of the relationship between the settlement and the sponsor. It's essentially the *right* over the governance of the settlement. 

The higher the points over a settlement, the more likely the sponsor will respond favorably to the need of the settlement.

> Maintains relations between other sponsors - e.g. SpaceX could cooperate with NASA or ESA, but also break the ties should this no longer be in agreement with their strategy.

Yes, to make things more complicated in the world of politics, I guess we could let the commander decide if he would let a *different* sponsor to pour in a large sum of resources over his colony. In doing so will definitely change the dynamic with the old sponsor.

**Comment by mokun** [comment]
> Maybe we could allow sponsors to buy settlements or installations on Mars from other sponsors. Or trade shares of a commercial base.

Will this go toward building a Mars monetary system or even a stock exchange ? :D 

That would be exciting to see money flowing in since it costs things to colonize the new world.

**Comment by Urwumpe** [comment]
Well, I just think in terms of story telling, a bit of Cyberpunk, a bit of classic science fiction. 

For example, allowing sponsors to buy and invest into settlements could also allow plots of the discovery of a huge mining resource near a base explosively increasing their value by many sponsors and cooperations trying to get access to that resource. 

I would just maybe not make the sponsors themselves acting and intelligent, but rather have "off-map" settlers, that don't live on Mars but otherwise behave just like martian settlers do. This way we can reuse a lot of code by having those sponsor management use the task system of martian settlers. For example, evaluating the value of a mining resource would be a task we need on Mars as well.

**Comment by mokun** [comment]
>  This way we can reuse a lot of code by having those sponsor management use the task system of martian settlers. 

But you will probably agree that the `Person` class now is way too complicated for sponsors to extend. 

Q: is there anyway to put the [ReportingAuthority](https://github.com/mars-sim/mars-sim/blob/master/mars-sim-core/src/main/java/org/mars_sim/msp/core/reportingAuthority/ReportingAuthority.java) into use ?

Right now, it's pretty hollow and have a method called `conductMissionObjective()` inside for setting the so-called *Mission Agenda*. 

However, I haven't quite got to the point of wiring up how the [MissionAgenda](https://github.com/mars-sim/mars-sim/blob/master/mars-sim-core/src/main/java/org/mars_sim/msp/core/reportingAuthority/MissionAgenda.java) can be used yet.

If not, we can always create a new `Sponsor` class to start afresh :D

**Comment by Urwumpe** [comment]
I agree, the Person class is a bit fat. Maybe we can clean it up a bit and delegate some parts of it into modules. Also I think we could maybe separate data model and calculations (For example, have a Person class to hold the data and a PersonWorker class that does the calculations in a thread pool), so we can easier serialize the Person into JSON or back.

Also, if I think ReportingAuthority is nicer as name, since it is not limited to a pure sponsorship relation. But we would still need some supporting framework below, so ReportingAuthority is really just a high-level object controlling multiple lower level objects for specific tasks - like having an organisation below it or persons acting. 

Essentially, since most organisations have a specific purpose, a Sponsor would really be only slightly different to any other organisation from this perspective.
